go/store/skip/list.go

```diff
@@ -15,64 +15,77 @@
 package skip
 
 import (
+	"hash/maphash"
 	"math"
-	"math/rand"
-
-	"github.com/zeebo/xxh3"
 )
 
 const (
-	maxCount = math.MaxUint32 - 1
-
-	maxHeight = uint8(5)
-	highest   = maxHeight - 1
-
+	maxHeight  = 9
+	maxCount   = math.MaxUint32 - 1
 	sentinelId = nodeId(0)
 )
 
+// A KeyOrder determines the ordering of two keys |l| and |r|.
+type KeyOrder func(l, r []byte) (cmp int)
+
+// A SeekFn facilitates seeking into a List. It returns true
+// if the seek operation should advance past |key|.
+type SeekFn func(key []byte) (advance bool)
+
+// List is an in-memory skip-list.
 type List struct {
+	// nodes contains all skipNode's in the List.
+	// skipNode's are assigned ascending id's and
+	// are stored in the order they were created,
+	// i.e. skipNode.id stores its index in |nodes|
 	nodes []skipNode
+
+	// count stores the current number of items in
+	// the list (updates are not made in-place)
 	count uint32
 
+	// checkpoint stores the nodeId of the last
+	// checkpoint made. All nodes created after this
+	// point will be discarded on a Revert()
 	checkpoint nodeId
-	cmp        ValueCmp
-	salt       uint64
-}
 
-type ValueCmp func(left, right []byte) int
+	// keyOrder determines the ordering of items
+	keyOrder KeyOrder
 
-type SearchFn func(nodeKey []byte) bool
+	// seed is hash salt
+	seed maphash.Seed
+}
 
 type nodeId uint32
 
-type skipPointer [maxHeight]nodeId
+type tower [maxHeight + 1]nodeId
```

Comment:
tower might be a good candidate for a docstring, something about how it is a stack of references to increasingly distant nodes in the list, with the distances inversely proportional to the probabilities in the init (or something else if i butchered that)

Reply:
good call, will add